src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -95,9 +95,14 @@ internal sealed class SourceMemberMethodSymbol : SourceMethodSymbol
 
             this.MakeFlags(methodKind, declarationModifiers, returnsVoid, isExtensionMethod, isMetadataVirtualIgnoringModifiers);
 
-            _typeParameters = (syntax.Arity == 0) ?
-                ImmutableArray<TypeParameterSymbol>.Empty :
-                MakeTypeParameters(syntax, diagnostics);
+            _typeParameters = syntax.Arity == 0
+                ? ImmutableArray<TypeParameterSymbol>.Empty
+                : MakeTypeParameters(syntax, diagnostics);
```

Comment:
Please merge this logic into the 'if' below so that we wouldn't have to check syntax.Arity twice.